src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/FeatureCheckDataFlow.cs

```diff
@@ -0,0 +1,999 @@
+				RequiresUnreferencedCode ();
+			}
+
+			[ExpectedWarning ("IL3050", nameof (RequiresDynamicCode), ProducedBy = Tool.Analyzer | Tool.NativeAot)]
+			[ExpectedWarning ("IL2026", nameof (RequiresUnreferencedCode))]
+			static void NotOr ()
+			{
+				if (!(RuntimeFeature.IsDynamicCodeSupported || TestFeatures.IsUnreferencedCodeSupported))
+					throw null;
+
+				RequiresDynamicCode ();
+				RequiresUnreferencedCode ();
+			}
+
+			static void EqualsTrue ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported == true)
+					RequiresDynamicCode ();
+			}
+
+			static void TrueEquals ()
+			{
+				if (true == RuntimeFeature.IsDynamicCodeSupported)
+					RequiresDynamicCode ();
+			}
+
+			static void EqualsFalse ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported == false)
+					throw null;
+
+				RequiresDynamicCode ();
+			}
+
+			static void FalseEquals ()
+			{
+				if (false == RuntimeFeature.IsDynamicCodeSupported)
+					throw null;
+
+				RequiresDynamicCode ();
+			}
+
+			static void NotEqualsTrue ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported != true)
+					throw null;
+					
+				RequiresDynamicCode ();
+			}
+
+			static void NotEqualsFalse ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported != false)
+					RequiresDynamicCode ();
+			}
+
+			static void TrueNotEquals ()
+			{
+				if (true != RuntimeFeature.IsDynamicCodeSupported)
+					throw null;
+
+				RequiresDynamicCode ();
+			}
+
+			static void FalseNotEquals ()
+			{
+				if (false != RuntimeFeature.IsDynamicCodeSupported)
+					RequiresDynamicCode ();
+			}
+
+			static void IsTrue ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported is true)
+					RequiresDynamicCode ();
+			}
+
+			static void IsFalse ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported is false)
+					throw null;
+
+				RequiresDynamicCode ();
+			}
+
+			static void IsNotTrue ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported is not true)
+					throw null;
+
+				RequiresDynamicCode ();
+			}
+
+			static void IsNotFalse ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported is not false)
+					RequiresDynamicCode ();
+			}
+
+			[ExpectedWarning ("IL3050", nameof (RequiresDynamicCode), ProducedBy = Tool.Analyzer | Tool.NativeAot)]
+			[ExpectedWarning ("IL3050", nameof (RequiresDynamicCode), ProducedBy = Tool.NativeAot)]
+			static void Contradiction ()
+			{
+				if (RuntimeFeature.IsDynamicCodeSupported && !RuntimeFeature.IsDynamicCodeSupported) {
+					RequiresDynamicCode ();
+				} else {
+					RequiresDynamicCode ();
+				}
+			}
+
+			public static void Test ()
+			{
+				And ();
+				AndNot ();
+				NotAnd ();
+				Or ();
+				OrNot ();
+				NotOr ();
+				EqualsTrue ();
+				TrueEquals ();
+				EqualsFalse ();
+				FalseEquals ();
+				NotEqualsTrue ();
+				NotEqualsFalse ();
+				TrueNotEquals ();
+				FalseNotEquals ();
+				IsTrue ();
+				IsFalse ();
+				IsNotTrue ();
+				IsNotFalse ();
+				Contradiction ();
+			}
+		}
+
+		class SupportedFeatureChecks
```

Comment:
Should this be called TestFeatureChecks (since it only validates the test-only stuff)